### PR TITLE
🚝 Monorepo!

### PR DESCRIPTION
**Pull request by jbrinley**
🚝 

Transforms tribe-libs into a monorepo. It can be included in Square One in whole, or sub-packages can be included individually.

To include the entire library:

```
composer require moderntribe/tribe-libs
```

Let's say you only want to use ACF:

```
composer require moderntribe/square1-acf
```

Thanks to composer dependencies, you'll also get `moderntribe/square1-object-meta` along with it.

This is definitely phase one of a larger transition. Most of the shared functionality we currently have in Square One can migrate into packages in Tribe Libs. That leaves the core plugin as a place to build genuinely project-specific logic.

For reference, here's the original document I sent around proposing this feature: https://docs.google.com/document/d/1ghExyuECI6tflooj53ibILVZ17TpOR0PDJibLllFuzY/edit

### CODE REVIEW

File: monorepo-builder.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+parameters:
+  package_directories:
+    - 'src'
```
Comment (Luc45) [inline monorepo-builder.yml]: Since these are now _third-party_ code, perhaps they could live in `packages` instead of `src`?

Comment (jbrinley) [reply monorepo-builder.yml]: I'm not sure what you mean by "third-party code". I have no strong preferences about the directory name.

Comment (Luc45) [reply monorepo-builder.yml]: I might be misunderstanding it - Is this thought correct?

- I clone a fresh Square One project
- `plugins/core/src` folder contains only code related to that specific project
- I require a package from the monorepo
- Now `plugins/core/src` folder also contains that monorepo package, which is tribe lib's code, not the project's

Comment (jbrinley) [reply monorepo-builder.yml]: Nope. `plugins/core/src` still contains the project code. The vendor code (i.e., what you install from our mini-repos) goes to `vendor/moderntribe/square1-*`

Comment (Luc45) [reply monorepo-builder.yml]: :aw-yeah: thanks for the clarification

**Comment by Luc45** [comment]
This thing is awesome :shipit: I would like to review my understanding of it, just to make sure that I'm not making too many wrong assumptions:

- A project would **usually** require a `component` _(or whathever a mini-repo is called?)_ with a major version constraint, like: `composer require moderntribe/square1-acf "^3.0"`
- `square1-acf` will automatically handle its own dependencies, including other `components`
- We would work on the project, assuming `^3.0` works as we expect and the API won't introduce breaking changes, including the dependencies
- Say we found a security issue on `square1-acf` and submitted a fix, bumping a patch version, say, to `3.0.1`
- We just have to run `composer update` on the project to have the fix available

PS: About versioning, are we considering following any versioning convention for the `components`? Like [semver](https://semver.org/) or similar

**Comment by jbrinley** [comment]
All of that was correct, @Luc45. Yes, tribe libs should use semver.

> We would work on the project, assuming ^3.0 works as we expect and the API won't introduce breaking changes, including the dependencies

Yes. Also, if one of the dependencies _does_ introduce a breaking change, that will cause the entire tribe-libs monorepo to bump its major version, which will then be distributed to the packages.

**Pull request merged**